From 315da2e6c219402f9d74305e314f4fcc2c4249c4 Mon Sep 17 00:00:00 2001
From: Semih Yavuz <semih.yavuz@qt.io>
Date: Tue, 16 Sep 2025 13:13:56 +0200
Subject: [PATCH 039/239] Add security header for  src/qmldom

There is no security critic code in qmldom. Everything remains
at the default "signicant" security level. Exception is one file
containing only headers marked "insignificant".

QUIP: 23
Fixes: QTBUG-136197
Pick-to: 6.9 6.8
Change-Id: Ied3f8f9299b8c429e903922b815ee5db8f2d1a9e
Reviewed-by: Ulf Hermann <ulf.hermann@qt.io>
(cherry picked from commit 14785c28679923e82f9047331235b6181609bad1)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qmldom/qqmldom_fwd_p.h           | 1 +
 src/qmldom/qqmldom_global.h          | 1 +
 src/qmldom/qqmldom_utils.cpp         | 1 +
 src/qmldom/qqmldom_utils_p.h         | 1 +
 src/qmldom/qqmldomastcreator.cpp     | 1 +
 src/qmldom/qqmldomastcreator_p.h     | 1 +
 src/qmldom/qqmldomastdumper.cpp      | 1 +
 src/qmldom/qqmldomastdumper_p.h      | 1 +
 src/qmldom/qqmldomcodeformatter.cpp  | 1 +
 src/qmldom/qqmldomcodeformatter_p.h  | 1 +
 src/qmldom/qqmldomcomments.cpp       | 1 +
 src/qmldom/qqmldomcomments_p.h       | 1 +
 src/qmldom/qqmldomcompare.cpp        | 2 ++
 src/qmldom/qqmldomcompare_p.h        | 1 +
 src/qmldom/qqmldomconstants.cpp      | 2 ++
 src/qmldom/qqmldomconstants_p.h      | 1 +
 src/qmldom/qqmldomelements.cpp       | 2 ++
 src/qmldom/qqmldomelements_p.h       | 1 +
 src/qmldom/qqmldomerrormessage.cpp   | 2 ++
 src/qmldom/qqmldomerrormessage_p.h   | 1 +
 src/qmldom/qqmldomexternalitems_p.h  | 1 +
 src/qmldom/qqmldomscanner.cpp        | 1 +
 src/qmldom/qqmldomscriptelements.cpp | 1 +
 src/qmldom/qqmldomscriptelements_p.h | 1 +
 src/qmldom/qqmldomstringdumper.cpp   | 1 +
 src/qmldom/qqmldomstringdumper_p.h   | 1 +
 src/qmldom/qqmldomtop.cpp            | 1 +
 src/qmldom/qqmldomtop_p.h            | 1 +
 src/qmldom/qqmldomtypesreader.cpp    | 1 +
 src/qmldom/qqmldomtypesreader_p.h    | 1 +
 30 files changed, 34 insertions(+)

diff --git a/src/qmldom/qqmldom_fwd_p.h b/src/qmldom/qqmldom_fwd_p.h
index 6698ae70e4..eaf8c83a89 100644
--- a/src/qmldom/qqmldom_fwd_p.h
+++ b/src/qmldom/qqmldom_fwd_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLDOM_FWD_P_H
 #define QQMLDOM_FWD_P_H
diff --git a/src/qmldom/qqmldom_global.h b/src/qmldom/qqmldom_global.h
index e0d343ab44..24035c1858 100644
--- a/src/qmldom/qqmldom_global.h
+++ b/src/qmldom/qqmldom_global.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:insignificant
 
 #ifndef QMLDOM_GLOBAL_H
 #define QMLDOM_GLOBAL_H
diff --git a/src/qmldom/qqmldom_utils.cpp b/src/qmldom/qqmldom_utils.cpp
index a3e73ac3ef..8eb7336a91 100644
--- a/src/qmldom/qqmldom_utils.cpp
+++ b/src/qmldom/qqmldom_utils.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmldom_utils_p.h"
 #include <QtCore/qdir.h>
diff --git a/src/qmldom/qqmldom_utils_p.h b/src/qmldom/qqmldom_utils_p.h
index c4cb8b0c16..827d05453a 100644
--- a/src/qmldom/qqmldom_utils_p.h
+++ b/src/qmldom/qqmldom_utils_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLDOM_UTILS_P_H
 #define QQMLDOM_UTILS_P_H
diff --git a/src/qmldom/qqmldomastcreator.cpp b/src/qmldom/qqmldomastcreator.cpp
index c36acd1a8f..04a99044ba 100644
--- a/src/qmldom/qqmldomastcreator.cpp
+++ b/src/qmldom/qqmldomastcreator.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmldomastcreator_p.h"
 #include "qqmldomconstants_p.h"
diff --git a/src/qmldom/qqmldomastcreator_p.h b/src/qmldom/qqmldomastcreator_p.h
index 1092535c6d..c4b1ab1071 100644
--- a/src/qmldom/qqmldomastcreator_p.h
+++ b/src/qmldom/qqmldomastcreator_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLDOMASTCREATOR_P_H
 #define QQMLDOMASTCREATOR_P_H
diff --git a/src/qmldom/qqmldomastdumper.cpp b/src/qmldom/qqmldomastdumper.cpp
index 0a8fb17329..8fb1c2fded 100644
--- a/src/qmldom/qqmldomastdumper.cpp
+++ b/src/qmldom/qqmldomastdumper.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmldomastdumper_p.h"
 #include "qqmldomerrormessage_p.h"
diff --git a/src/qmldom/qqmldomastdumper_p.h b/src/qmldom/qqmldomastdumper_p.h
index 542f81b1dd..ecbb783806 100644
--- a/src/qmldom/qqmldomastdumper_p.h
+++ b/src/qmldom/qqmldomastdumper_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLDOMASTDUMPER_P_H
 #define QQMLDOMASTDUMPER_P_H
diff --git a/src/qmldom/qqmldomcodeformatter.cpp b/src/qmldom/qqmldomcodeformatter.cpp
index a809361435..017cef80b5 100644
--- a/src/qmldom/qqmldomcodeformatter.cpp
+++ b/src/qmldom/qqmldomcodeformatter.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2022 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmldomcodeformatter_p.h"
 
diff --git a/src/qmldom/qqmldomcodeformatter_p.h b/src/qmldom/qqmldomcodeformatter_p.h
index 084272401e..1d8b3a31e3 100644
--- a/src/qmldom/qqmldomcodeformatter_p.h
+++ b/src/qmldom/qqmldomcodeformatter_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2022 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLDOMCODEFORMATTER_P_H
 #define QQMLDOMCODEFORMATTER_P_H
diff --git a/src/qmldom/qqmldomcomments.cpp b/src/qmldom/qqmldomcomments.cpp
index 91e1153b49..bed0880d2f 100644
--- a/src/qmldom/qqmldomcomments.cpp
+++ b/src/qmldom/qqmldomcomments.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmldomcomments_p.h"
 #include "qqmldomoutwriter_p.h"
diff --git a/src/qmldom/qqmldomcomments_p.h b/src/qmldom/qqmldomcomments_p.h
index ebc51add04..2e77861127 100644
--- a/src/qmldom/qqmldomcomments_p.h
+++ b/src/qmldom/qqmldomcomments_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLDOMCOMMENTS_P_H
 #define QQMLDOMCOMMENTS_P_H
diff --git a/src/qmldom/qqmldomcompare.cpp b/src/qmldom/qqmldomcompare.cpp
index 6bd94d90de..5631ae9a51 100644
--- a/src/qmldom/qqmldomcompare.cpp
+++ b/src/qmldom/qqmldomcompare.cpp
@@ -1,5 +1,7 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
+
 #include "qqmldomcompare_p.h"
 #include "QtCore/qglobal.h"
 
diff --git a/src/qmldom/qqmldomcompare_p.h b/src/qmldom/qqmldomcompare_p.h
index 651486d7a9..4e7cd053b7 100644
--- a/src/qmldom/qqmldomcompare_p.h
+++ b/src/qmldom/qqmldomcompare_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QMLDOMCOMPARE_P_H
 #define QMLDOMCOMPARE_P_H
diff --git a/src/qmldom/qqmldomconstants.cpp b/src/qmldom/qqmldomconstants.cpp
index e6068fdcbb..0dbdc51378 100644
--- a/src/qmldom/qqmldomconstants.cpp
+++ b/src/qmldom/qqmldomconstants.cpp
@@ -1,5 +1,7 @@
 // Copyright (C) 2022 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:insignificant
+
 #include "qqmldomconstants_p.h"
 
 QT_BEGIN_NAMESPACE
diff --git a/src/qmldom/qqmldomconstants_p.h b/src/qmldom/qqmldomconstants_p.h
index 1b0ce8273a..2f6ec1510b 100644
--- a/src/qmldom/qqmldomconstants_p.h
+++ b/src/qmldom/qqmldomconstants_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLDOMCONSTANTS_P_H
 #define QQMLDOMCONSTANTS_P_H
diff --git a/src/qmldom/qqmldomelements.cpp b/src/qmldom/qqmldomelements.cpp
index 35defa7036..443566ea43 100644
--- a/src/qmldom/qqmldomelements.cpp
+++ b/src/qmldom/qqmldomelements.cpp
@@ -1,5 +1,7 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
+
 // Suppress GCC 11 warning about maybe-uninitialized copy of
 // another Data. We're not sure if the compiler is actually right,
 // but in this type of warning, it often isn't.
diff --git a/src/qmldom/qqmldomelements_p.h b/src/qmldom/qqmldomelements_p.h
index c6781c6ab8..430cd34eb9 100644
--- a/src/qmldom/qqmldomelements_p.h
+++ b/src/qmldom/qqmldomelements_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLDOMELEMENTS_P_H
 #define QQMLDOMELEMENTS_P_H
diff --git a/src/qmldom/qqmldomerrormessage.cpp b/src/qmldom/qqmldomerrormessage.cpp
index 032cebe817..f4b2dc8c74 100644
--- a/src/qmldom/qqmldomerrormessage.cpp
+++ b/src/qmldom/qqmldomerrormessage.cpp
@@ -1,5 +1,7 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
+
 #include "qqmldomitem_p.h"
 #include "qqmldomstringdumper_p.h"
 #include "qqmldomfilelocations_p.h"
diff --git a/src/qmldom/qqmldomerrormessage_p.h b/src/qmldom/qqmldomerrormessage_p.h
index a70cb4e838..4198d31f8a 100644
--- a/src/qmldom/qqmldomerrormessage_p.h
+++ b/src/qmldom/qqmldomerrormessage_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef ERRORMESSAGE_H
 #define ERRORMESSAGE_H
diff --git a/src/qmldom/qqmldomexternalitems_p.h b/src/qmldom/qqmldomexternalitems_p.h
index b61ca75307..fbb4ddb3cc 100644
--- a/src/qmldom/qqmldomexternalitems_p.h
+++ b/src/qmldom/qqmldomexternalitems_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLDOMEXTERNALITEMS_P_H
 #define QQMLDOMEXTERNALITEMS_P_H
diff --git a/src/qmldom/qqmldomscanner.cpp b/src/qmldom/qqmldomscanner.cpp
index 68dcb7c087..463a5dc060 100644
--- a/src/qmldom/qqmldomscanner.cpp
+++ b/src/qmldom/qqmldomscanner.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2022 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmldomscanner_p.h"
 #include "qqmldomerrormessage_p.h"
diff --git a/src/qmldom/qqmldomscriptelements.cpp b/src/qmldom/qqmldomscriptelements.cpp
index 768df84f34..e68a2da96f 100644
--- a/src/qmldom/qqmldomscriptelements.cpp
+++ b/src/qmldom/qqmldomscriptelements.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmldom_utils_p.h"
 #include "qqmldomitem_p.h"
diff --git a/src/qmldom/qqmldomscriptelements_p.h b/src/qmldom/qqmldomscriptelements_p.h
index 0148fc3f1a..c1153d46e3 100644
--- a/src/qmldom/qqmldomscriptelements_p.h
+++ b/src/qmldom/qqmldomscriptelements_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2023 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLDOMSCRIPTELEMENTS_P_H
 #define QQMLDOMSCRIPTELEMENTS_P_H
diff --git a/src/qmldom/qqmldomstringdumper.cpp b/src/qmldom/qqmldomstringdumper.cpp
index 16c2cfe688..0a56ecb1de 100644
--- a/src/qmldom/qqmldomstringdumper.cpp
+++ b/src/qmldom/qqmldomstringdumper.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmldomstringdumper_p.h"
 #include <QtCore/QDebug>
diff --git a/src/qmldom/qqmldomstringdumper_p.h b/src/qmldom/qqmldomstringdumper_p.h
index cf5c8e6483..e4db880d54 100644
--- a/src/qmldom/qqmldomstringdumper_p.h
+++ b/src/qmldom/qqmldomstringdumper_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef DUMPER_H
 #define DUMPER_H
diff --git a/src/qmldom/qqmldomtop.cpp b/src/qmldom/qqmldomtop.cpp
index c784635df1..a6f3149e71 100644
--- a/src/qmldom/qqmldomtop.cpp
+++ b/src/qmldom/qqmldomtop.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmldomitem_p.h"
 #include "qqmldomtop_p.h"
diff --git a/src/qmldom/qqmldomtop_p.h b/src/qmldom/qqmldomtop_p.h
index 1e9485a933..800b4c8b68 100644
--- a/src/qmldom/qqmldomtop_p.h
+++ b/src/qmldom/qqmldomtop_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef DOMTOP_H
 #define DOMTOP_H
diff --git a/src/qmldom/qqmldomtypesreader.cpp b/src/qmldom/qqmldomtypesreader.cpp
index 7bf0223770..6820b8a4d6 100644
--- a/src/qmldom/qqmldomtypesreader.cpp
+++ b/src/qmldom/qqmldomtypesreader.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmldomtypesreader_p.h"
 #include "qqmldomelements_p.h"
diff --git a/src/qmldom/qqmldomtypesreader_p.h b/src/qmldom/qqmldomtypesreader_p.h
index dda0b8f5b3..cef833c9da 100644
--- a/src/qmldom/qqmldomtypesreader_p.h
+++ b/src/qmldom/qqmldomtypesreader_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2019 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLDOMTYPESREADER_H
 #define QQMLDOMTYPESREADER_H
-- 
2.51.2

